doctr/models/resnet.py

```diff
@@ -0,0 +1,138 @@
+
+    @staticmethod
+    def conv_resnetblock(output_channels, kernel_size):
+        return [
+            layers.Conv2D(output_channels, kernel_size, padding='same', use_bias=False, kernel_initializer='he_normal'),
+            layers.BatchNormalization(),
+            layers.Activation('relu'),
+            layers.Conv2D(output_channels, kernel_size, padding='same', use_bias=False, kernel_initializer='he_normal'),
+            layers.BatchNormalization(),
+        ]
+
+    def call(
+        self,
+        inputs: tf.Tensor
+    ) -> tf.Tensor:
+        clone = self.shortcut(inputs)
+        conv_out = self.conv_block(inputs)
+        out = self.act(clone + conv_out)
+
+        return out
+
+
+class ResnetStage(Sequential):
+    """Implements a resnet31 stage
+
+    Args:
+        num_blocks: number of blocks inside the stage
+        output_channels: number of channels to use in Conv2D
+
+    """
+    def __init__(
+        self,
+        num_blocks: int,
+        output_channels: int
+    ) -> None:
+
+        super().__init__()
+        final_blocks = [
+            ResnetBlock(output_channels, conv_shortcut=False) for _ in range(1, num_blocks)
+        ]
+        self.add(ResnetBlock(output_channels, conv_shortcut=True))
+        for final_block in final_blocks:
+            self.add(final_block)
+
+
+class Resnet31(Sequential):
+    """Resnet31 architecture with rectangular pooling windows as described in
+    `"Show, Attend and Read:A Simple and Strong Baseline for Irregular Text Recognition",
+    <https://arxiv.org/pdf/1811.00751.pdf>`_. Downsizing: (H, W) --> (H/8, W/4)
+
+    Args:
+        input_size: size of the images
+
+    """
+    def __init__(
+        self,
+        input_size: Tuple[int, int, int] = (640, 640, 3)
+    ) -> None:
+
+        _layers = [
+            *self.conv_bn_act(output_channels=64, kernel_size=3, input_shape=input_size, use_bias=False),
+            *self.conv_bn_act_pool(output_channels=128, kernel_size=3, p_size=2, use_bias=False),
+            ResnetStage(num_blocks=1, output_channels=256),
+            *self.conv_bn_act_pool(output_channels=256, kernel_size=3, p_size=2, use_bias=False),
+            ResnetStage(num_blocks=2, output_channels=256),
+            *self.conv_bn_act_pool(output_channels=256, kernel_size=3, p_size=(2, 1), use_bias=False),
+            ResnetStage(num_blocks=5, output_channels=512),
+            *self.conv_bn_act(output_channels=512, kernel_size=3, use_bias=False),
+            ResnetStage(num_blocks=3, output_channels=512),
+            *self.conv_bn_act(output_channels=512, kernel_size=3, use_bias=False),
+        ]
+        super().__init__(_layers)
+
+    @staticmethod
+    def conv_bn_act(output_channels, kernel_size, **kwargs):
+        return [
+            layers.Conv2D(output_channels, kernel_size, padding='same', kernel_initializer='he_normal', **kwargs),
+            layers.BatchNormalization(),
+            layers.Activation('relu'),
+        ]
```

Comment:
You can use `conv_sequence` from `doctr.models.utils` which is achieving the same thing :ok_hand: